File: BOJ/JavaScript/10773.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

class Stack {
  constructor() {
    this.items = [];
  }

  push(el) {
    this.items.push(el);
  }

  pop() {
    if (!this.empty()) {
      return this.items.pop();
    }
  }

  empty() {
    return this.items.length === 0;
  }

  size() {
    return this.items.length;
  }

  top() {
    return this.items[this.items.length - 1];
  }
}

input.shift();
const arr = input.map((v) => +v);
const stack = new Stack();
let ret = 0;

arr.forEach((v) => (v ? stack.push(v) : !stack.empty() && stack.pop()));

while (!stack.empty()) {
  ret += stack.top();
  stack.pop();
}

console.log(ret);

File: BOJ/10000/10773.cpp
#include <bits/stdc++.h>
using namespace std;

int k, a, ret;
stack<int> st;

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  cin >> k;

  while (k--) {
    cin >> a;

    if (a) {
      st.push(a);
    } else {
      if (st.size() != 0) st.pop();
    }
  }

  while (st.size() != 0) {
    ret += st.top();
    st.pop();
  }

  cout << ret;

  return 0;
}